asdf/tests/test_api.py

```diff
@@ -82,6 +84,27 @@ def test_open_readonly(tmpdir):
             pass
 
 
+def test_open_validate_on_read(tmpdir):
+    tmpfile = str(tmpdir.join('invalid.asdf'))
```

Comment:
Is there ever anything written to this file path? I don't see it in the code below, but maybe there is some test magic going on that isn't apparent.

Reply:
Uhhhhh... _waves hand vaguely_ this is not the tempfile you're looking for